clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2557,7 +2557,10 @@ boolean handleUnsupportedVersionException(final UnsupportedVersionException exce
                     return false;
                 }
 
-                if (exception.getMessage().contains("Including fenced broker endpoints is not supported with version")) {
+                // If unsupportedVersion exception was caused by the option to include fenced brokers (only supported for version 1+)
+                // then we should not fall back to the metadataRequest.
+                if (options.includeFencedBrokers()) {
+                    log.error("Including fenced broker endpoints is not supported with " + exception.getMessage());
```

Comment:
I'm not sure the exception message when unsupportedVersionException thrown. You need to check again in integration test. The `Api DESCRIBE_CLUSTER with version 1` is from `mockClient`, not real `networkClient`. FYI.

Reply:
Right. The actual error message you get is `Attempted to write a non-default includeFencedBrokers at version 1`. Perhaps this is self explanatory so don't need this log?

Reply:
I think so.